ludwig/utils/image_utils.py

```diff
@@ -163,6 +163,71 @@ def read_image_from_str(img: str, num_channels: Optional[int] = None) -> torch.T
         return None
 
 
+def get_image_read_mode_from_num_channels(num_channels: int) -> ImageReadMode:
+    """Returns the torchvision.io.ImageReadMode corresponding to the number of channels.
+
+    If num_channels is not recognized, returns ImageReadMode.UNCHANGED.
+    """
+    mode = ImageReadMode.UNCHANGED
+    if num_channels == 1:
+        mode = ImageReadMode.GRAY
+    elif num_channels == 2:
+        mode = ImageReadMode.GRAY_ALPHA
+    elif num_channels == 3:
+        mode = ImageReadMode.RGB
+    elif num_channels == 4:
+        mode = ImageReadMode.RGB_ALPHA
+    return mode
+
+
+def read_image_if_path(path: Any, num_channels: Optional[int] = None) -> Union[Any, torch.Tensor]:
+    """Gets an image if `path` is a path (e.g. a string).
+
+    If it is not a path, return as-is.
+    """
+    if not isinstance(path, str):
+        return path
+    return read_image_from_path(path, num_channels)
+
+
+def read_image_from_path(path: str, num_channels: Optional[int] = None) -> Optional[torch.Tensor]:
+    """Reads image from path.
+
+    Useful for reading from a small number of paths. For more intensive reads, use backend.read_binary_files instead.
+    """
+    bytes_obj = get_bytes_obj_from_path(path)
+    return read_image_from_bytes_obj(bytes_obj, num_channels)
+
+
+def read_image_if_bytes_obj(
+    bytes_obj: Optional[bytes] = None, num_channels: Optional[int] = None
+) -> Union[Any, Optional[torch.Tensor]]:
+    """Reads image if `bytes_obj` is a bytes object.
+
+    If it is not a bytes object, return as-is.
+    """
+    if not isinstance(bytes_obj, bytes):
+        return bytes_obj
+    return read_image_from_bytes_obj(bytes_obj, num_channels)
+
+
+def read_image_from_bytes_obj(
+    bytes_obj: Optional[bytes] = None, num_channels: Optional[int] = None
+) -> Optional[torch.Tensor]:
+    mode = get_image_read_mode_from_num_channels(num_channels)
+
+    try:
+        with BytesIO(bytes_obj) as buffer:
+            buffer_view = buffer.getbuffer()
+            image = decode_image(torch.frombuffer(buffer_view, dtype=torch.uint8), mode=mode)
+            del buffer_view
```

Comment:
Why is it necessary to `del buffer_view`?  Seems like it will go out-of-scope on the next line.  If I'm missing something, please add a comment here explaining why this is needed.

Reply:
Hm, not quite sure why it's needed. This was a bit of code that was already in `read_image`: https://github.com/ludwig-ai/ludwig/blob/d3eea1347d394b363ebf2cdfe973bd9adbd84a76/ludwig/utils/image_utils.py#L121

Reply:
AFAICT it would be safe to remove this line.

Reply:
Hm, it seems like removing this is causing tests to fail with the following error:
```
Existing exports of data: object cannot be re-sized
```
Seems related to this open issue: https://github.com/python/cpython/issues/85269

I've put it back for now, but let me know if you think there's an alternative you think is worth exploring.